allennlp/models/encoder_decoders/wikitables_semantic_parser.py

```diff
@@ -1117,6 +1126,14 @@ def take_step(self,
             # entity_action_mask: (group_size, num_entity_actions)
             entity_action_logits, entity_action_mask, entity_type_embeddings = \
                     self._get_entity_action_logits(state, actions_to_link, attention_weights)
+
+            # The entity and action logits are combined with a mixture weight to prevent the
+            # entity_action_logits from dominating the embedded_action_logits if a softmax
+            # was applied on both together.
+            mixture_weight = self._mixture_feedforward(hidden_state)
+            entity_action_logits = util.masked_softmax(entity_action_logits, entity_action_mask) * mixture_weight
+            embedded_action_logits = util.masked_softmax(embedded_action_logits, embedded_action_mask) * mixture_weight
```

Comment:
This is close, but not quite right, because these aren't logits anymore, after you've done a softmax.  These are now probabilities.  So, you can't do another softmax with these below - some more logic needs to be modified to make this work right.

Reply:
done.